GitUI/UserControls/RevisionGrid/Graph/ThreadSafeGrowingList.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace GitUI.UserControls.RevisionGrid.Graph
+{
+    // This list can only grow. Because it only grows, reading is thread safe.
+    // Writing to this list is not thread safe. However, for the usecase
+    // where only one thread is adding items to the list, and multiple threads
+    // are reading, this can be used.
+    public class ThreadSafeGrowingList<T> : IReadOnlyList<T>
+    {
+        private T[] _internalArray;
+
+        public ThreadSafeGrowingList(int capacity = 2)
```

Comment:
📝 `Add` will fail if `capacity` is initialized as 0.

Reply:
Added a guard